reproject/adaptive/core.py

```diff
@@ -31,9 +31,11 @@ def __call__(self, pixel_out):
         return pixel_in
 
 
-def _reproject_adaptive_2d(array, wcs_in, wcs_out, shape_out, order=1,
```

Comment:
The removal of this order keyword is also a breaking change. What was it being used for before this PR? Can we deprecate it?

Reply:
That's an inconvenient but good point. The `order` argument was controlling a linear interpolation step being applied before the weighted averaging that's at the core of this algorithm. The linear step isn't present in the paper describing this algorithm, and it turns out to have subtle effects that reduce the quality of the reprojection, as described and demonstrated [here](https://gist.github.com/svank/17e3544f7d3079f2d9f11c9fe0a95bbf). It's an extra step with no upside that I'm aware of, so removing it completely makes sense to me.

One way to make it a non-breaking change might be to keep the `order` argument and just ignore it (and also raise a deprecation warning?). Outputs will also change slightly because of the removal of the linear interpolation step (not nearly as much as with a change to using the Gaussian kernel by default), but I'd put those changes in the "bug-fix" category. This PR and #274 both have other commits fixing bugs that will change outputs slightly.

Reply:
I would be happy with the kwarg still being there and just being ignored.

Reply:
I think raise a deprecation warning if order is anything other than the default value, as an approximation of user changed this, and then we can pull it in the subsequent release.